core/xchain/entitlement/check_operation.go

```diff
@@ -127,6 +143,47 @@ func (e *Evaluator) evaluateIsEntitledOperation(
 	return false, nil
 }
 
+// Check balance in Wei
+func (e *Evaluator) evaluateEthBalanceOperation(
+	ctx context.Context,
+	op *CheckOperation,
+	linkedWallets []common.Address,
+) (bool, error) {
+	log := dlog.FromCtx(ctx).With("function", "evaluateEthBalanceOperation")
+	client, err := e.clients.Get(op.ChainID.Uint64())
+	if err != nil {
+		log.Error("Chain ID not found", "chainID", op.ChainID)
+		return false, fmt.Errorf("evaluateEthBalanceOperation: Chain ID %v not found", op.ChainID)
+	}
+
+	total := big.NewInt(0)
+	for _, wallet := range linkedWallets {
+		// Balance is returned as a representation of the balance according the denomination of ETH.
+		// Default decimals for ETH is 18, meaning the balance is stored in Wei (1 ETH = 10^18 Wei).
+		balance, err := client.BalanceAt(ctx, wallet, nil)
+		if err != nil {
+			log.Error("Failed to retrieve ETH balance", "chain", op.ChainID, "error", err)
+			return false, err
+		}
+		total.Add(total, balance)
+
+		log.Info("Retrieved ETH balance",
+			"balance", balance.String(),
+			"total", total.String(),
+			"threshold", op.Threshold.String(),
+			"chainID", op.ChainID.String(),
+		)
+
+		// Balance is a *big.Int
+		// Iteratively check if the total balance of evaluated wallets is greater than or equal to the
+		// threshold.
+		if op.Threshold.Sign() > 0 && total.Sign() > 0 && total.Cmp(op.Threshold) >= 0 {
```

Comment:
Native coin balances can't be negative. So `op.Threshold` should always be >= 0, otherwise the request makes no sense. That also implies that a single check on `op.Threshold == 0` in the beginning of this function is enough. If it is 0 the result is a pass (if len(wallets) > 0). If it is negative the request is invalid because the coins native balances can't be negative.

Reply:
Actually NM, to this PR I added a threshold check to all checktypes for threshold > 0 and consider threshold == 0 invalid, since that's basically a no-op, along with unit tests. I'll refactor the comparisons too.